kafka/consumer.go

```diff
@@ -626,3 +643,37 @@ func (c *Consumer) SetOAuthBearerToken(oauthBearerToken OAuthBearerToken) error
 func (c *Consumer) SetOAuthBearerTokenFailure(errstr string) error {
 	return c.handle.setOAuthBearerTokenFailure(errstr)
 }
+
+// ConsumerGroupMetadata reflects the current consumer group member metadata.
+type ConsumerGroupMetadata struct {
+	cgmd *C.rd_kafka_consumer_group_metadata_t
+}
+
+// GetConsumerGroupMetadata returns the consumer's current group metadata.
+// This object should be passed to the transactional producer's
+// SendOffsetsToTransaction() API and then destroyed with metadata.Close().
+func (c *Consumer) GetConsumerGroupMetadata() *ConsumerGroupMetadata {
```

Comment:
It sounds strange that you need to do a `close()` on something you do `get()` to acquire. Getters usually return things owned by something else, instead of transferring ownership of it to the caller. What is the semantics of multiple `GetConsumerGroupMetadata()` calls? Would you need call `Close()` for each one? Do they return the same object (assuming nothing changed underneath)?

Reply:
Aye, fixed.